### PR TITLE
Refactor user controller

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
Includes the patch from #4630 since otherwise one of the tests would fail

File: src/api/app/controllers/webui/user_controller.rb [hunk 1]
```diff
@@ -102,6 +102,8 @@ def save
   def edit; end
 
   def update
+    params.permit(:user, :state, :make_admin)
```
Comment (hennevogel) [inline src/api/app/controllers/webui/user_controller.rb]: Please do this in the "standard" way. 

```ruby
    # Never trust parameters from the scary internet, only allow the white list through.
    def user_params
      params.require(:user).permit(:state, :make_admin)
    end
```

Comment (hennevogel) [reply src/api/app/controllers/webui/user_controller.rb]: Then here `@user.update(user_params)`

**Comment by bgeuken** [comment]
@hennevogel updated as well

**Comment by bgeuken** [comment]
[P4: (5) Allow local users in LDAP mode](https://trello.com/c/tj5UVn89/694-p4-5-allow-local-users-in-ldap-mode)

**Comment by hennevogel** [comment]
codeschnarch...

**Pull request merged**